prov/efa/src/efa_msg.c

```diff
@@ -123,6 +123,37 @@ static ssize_t efa_post_recv_validate(struct efa_ep *ep, const struct fi_msg *ms
 	return 0;
 }
 
+/**
+ * @brief convert rdma-core return code libfabric return code.
+ *
+ * On failure, rdma-core functions return positive errno, and
+ * libfabric functions return negative errno. This function does
+ * the conversion.
+ *
+ * Because libfabric errno is unix errno with some extensions
+ * (see include/rdma/fi_errno.h for libfabric errno). This function
+ * simply converts postive to negative, with one exception:
+ *
+ * If rdma-core return is ENOMEM, this function return -FI_EAGAIN.
+ *
+ * @param[in]	ibv_retcode	return code from rdma-core functions
+ * @return	return core for libfabric functions.
+ */
+static inline
+int efa_ibv_retcode_to_fi_retcode(int ibv_retcode)
```

Comment:
Gah, no, don't do this.  Handle this in the return code processing itself.  There's no reason for a function like this.

Reply:
Removed the function in new revision.